server.js

```diff
@@ -3,36 +3,100 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import topMusicData from './data/top-music.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8082
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+const Track = mongoose.model('Track', {
+  trackName: String,
+  artistName: String,
+  genre: String,
+  bpm: Number
+})
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Track.deleteMany({})
+
+    topMusicData.forEach((trackData) => {
+      new Track(trackData).save()
+    })
+  }
+  seedDatabase()
+}
+
+app.get('/', async (req, res) => {
+  const tracks = await Track.find()
+  res.json(tracks)
+})
+
+app.get('/tracks', async (req, res) => {
+  const genreQuery = new RegExp(req.query.genre, 'i')
+  const bpmQuery = req.query.bpm
+  let tracks = await Track.find()
+
+  if (genreQuery && bpmQuery) {
+    tracks = await Track.find({
+      $and: [{ genre: genreQuery }, { bpm: bpmQuery }]
+    })
+  } else if (genreQuery || bpm) {
+    tracks = await Track.find({
+      $or: [{ genre: genreQuery }, { bpm: bpmQuery }]
+    })
+  }
```

Comment:
Interesting logic.  I wonder what would happen if you included another field here.  How the if statement would look in that case.